supervisor/shared/rgb_led_status.c

```diff
@@ -386,20 +387,22 @@ void prep_rgb_status_animation(const pyexec_result_t* result,
     if (!status->ok) {
         status->total_exception_cycle = EXCEPTION_TYPE_LENGTH_MS * 3 + LINE_NUMBER_TOGGLE_LENGTH * status->digit_sum + LINE_NUMBER_TOGGLE_LENGTH * num_places;
     }
-    if (mp_obj_is_subclass_fast(result->exception_type, &mp_type_IndentationError)) {
-        status->exception_color = INDENTATION_ERROR;
-    } else if (mp_obj_is_subclass_fast(result->exception_type, &mp_type_SyntaxError)) {
-        status->exception_color = SYNTAX_ERROR;
-    } else if (mp_obj_is_subclass_fast(result->exception_type, &mp_type_NameError)) {
-        status->exception_color = NAME_ERROR;
-    } else if (mp_obj_is_subclass_fast(result->exception_type, &mp_type_OSError)) {
-        status->exception_color = OS_ERROR;
-    } else if (mp_obj_is_subclass_fast(result->exception_type, &mp_type_ValueError)) {
-        status->exception_color = VALUE_ERROR;
-    } else if (mp_obj_is_subclass_fast(result->exception_type, &mp_type_MpyError)) {
-        status->exception_color = MPY_ERROR;
-    } else {
-        status->exception_color = OTHER_ERROR;
+    if (result->exception_type) {
```

Comment:
How about making the first case the inverse of this and set the exception color to other in that case? Then you get an exception color and no if nesting.

Reply:
Was found_main true in this case? With code.py it should be false and we can just skip all of the exception flashing.

Reply:
I can check out found_main and see what the value is - I'm not that familiar with this portion of Circuitpython so I was kind of blundering around here. `exception_type` is the problematic element - it's initialized to NULL in main.c and never changed, so anything based off of it will fail. We could set it to a proper default instead, if you'd like - the problematic stuff starts [here](https://github.com/adafruit/circuitpython/blob/16170d44abc55d8a5e65b903fa3e200c0154cbc1/main.c#L247). Still not sure why STM and the other ports don't trigger a null reference crash too.

Reply:
What exactly is `found_main` supposed to do? I'm very confused by this whole section. Circuitpython runs with `start_mp`, then it tries to find the file *afterwards?* And you say that it should be false for `code.py`, but code.py is in the list of filenames it matches?